### PR TITLE
Enable PayPal as a payment method

### PR DESCRIPTION
**Pull request by luukveenis**
This PR is based entirely of #34 , with some additional refactoring. It adds PayPal as a payment option through the regular checkout flow if we set the payment method to display on the frontend, and also adds a PayPal OneTouch button on the cart page.

In the refactoring I tried to make it easy to create a PayPal button, but flexible enough that users can specify what type of button they want (checkout, vault, and other options). For example:
```
<script>
  var button = new PaypalButton(document.querySelector("#paypal-button");
  button.setTokenizeCallback(myCallbackFunction);
  button.initialize({
    flow: 'fault',
    enableShippingAddress: true
   });
</script>
```

### CODE REVIEW

**Comment by cbrunsdon** [comment]
This is good by me if we can green.

**Comment by luukveenis** [comment]
jenkins test this please

**Comment by luukveenis** [comment]
jenkins test this please

File: spec/models/solidus_paypal_braintree/source_spec.rb [hunk 1]
```diff
@@ -86,4 +86,56 @@
...
+    context "when the payment type is Apple Pay" do
+      let(:type) { "ApplePayCard" }
+
+      it "returns the translated payment type" do
+        expect(subject).to eq "Apple Pay"
+      end
+    end
+  end
+
+  describe "#apple_pay??" do
```
Comment (adammathys) [inline spec/models/solidus_paypal_braintree/source_spec.rb]: `#apple_pay??`

**Comment by luukveenis** [comment]
jenkins test this please

**Comment by luukveenis** [comment]
jenkins test this please

**Pull request merged**